static/src/js/components/SpatialDisplay/SpatialDisplay.jsx

```diff
@@ -665,10 +694,11 @@ const SpatialDisplay = ({
       />
     ))
   } else if (
-    (
+    ((
       (currentPolygonSearch && currentPolygonSearch.length) && !drawingNewLayer
     )
-    || drawingNewLayer === spatialTypes.POLYGON
+    || drawingNewLayer === spatialTypes.POLYGON)
+    && !(nlpCollection && nlpCollection.spatial && nlpCollection.spatial.geoLocation) // Don't show polygon section when NLP data is present
```

Comment:
Polygon search shouldn't be populated if the nlp flag is true, so this is unnecessary